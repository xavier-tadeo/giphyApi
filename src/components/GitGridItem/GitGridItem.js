import React from "react";
import "./GitGridItem.scss";

export const GitGridItem = ({ id, img, title }) => {
  return (
    <div className="card animate__animated animate__zoomInDown">
      <p className="card__title">{title}</p>
      <img className="card__image" src={img} alt={title} />
    </div>
  );
};
