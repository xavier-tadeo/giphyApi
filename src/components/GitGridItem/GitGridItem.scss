@import url("https://fonts.googleapis.com/css2?family=Licorice&display=swap");

.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 10px;
  &__title {
    margin: 0;
    padding: 10px;
    font-size: 16px;
    font-family: "Licorice", cursive;
  }

  &__image {
    height: 200px;
    width: 250px;
    padding: 5px 0;
    border-radius: 50px;
  }
}
