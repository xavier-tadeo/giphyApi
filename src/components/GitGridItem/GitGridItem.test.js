import { shallow } from "enzyme";
import { GitGridItem } from "./GitGridItem";

describe("Given a GitGridItem component", () => {
  describe("When it render", () => {
    test("Then should rendered a pTag and imgTag", () => {
      const image = "url";
      const title = "Un titulo";

      const wrapper = shallow(<GitGridItem img={image} title={title} />);
      const pExpect = wrapper.find("p").text();

      expect(pExpect).toBe(title);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
