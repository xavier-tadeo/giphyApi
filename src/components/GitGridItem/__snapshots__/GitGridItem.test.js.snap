// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a GitGridItem component When it render Then should rendered a pTag and imgTag 1`] = `
<div
  className="card animate__animated animate__zoomInDown"
>
  <p
    className="card__title"
  >
    Un titulo
  </p>
  <img
    alt="Un titulo"
    className="card__image"
    src="url"
  />
</div>
`;
